perf: Maps for app categories

keithwillcode opened the pull request:
## What does this PR do?

Similar to what was done in #19718, we now have a smaller map of payment apps to avoid importing the entire app store for just a few payment apps.

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Ensure payment apps are still working

keithwillcode committed:
perf: Add payment apps map

keithwillcode committed:
Added a calendar apps map

keithwillcode committed:
Added a video adapter map

keithwillcode committed:
Fixed types

keithwillcode committed:
All imports of full app-store are gone

keithwillcode committed:
Renamed calendar.apps to calendar.services

keithwillcode committed:
Trying to fix app-store mocks

keithwillcode committed:
ignoring warnings

keithwillcode committed:
perf: Broke the apps.browser.generated.tsx file into multiple files

Removed apps.browser.generated.tsx file

keithwillcode committed:
Removed app-store index.ts since it's no longer used

keithwillcode committed:
Reverted unwanted change

keithwillcode commented on apps/web/next.config.js:
```diff
@@ -225,6 +225,11 @@ const nextConfig = {
     unoptimized: true,
   },
   webpack: (config, { webpack, buildId, isServer }) => {
+    config.ignoreWarnings = [
```
Just for testing. With this error, Next.js doesn't always show you the number of modules compiled.

keithwillcode committed:
Update tsconfig.json

keithwillcode commented on packages/lib/tsconfig.json:
```diff
@@ -6,5 +6,5 @@
   },
   "include": [".", "../types/next-auth.d.ts", "../types/window.d.ts", "../trpc/types/router.d.ts"],
-  "exclude": ["dist", "build", "**/node_modules/**"]
+  "exclude": ["dist", "build", "**/node_modules/**", "**/trpc/**"]
```
This should be reverted before merge

keithwillcode committed:
Fix some types

keithwillcode committed:
Merge branch 'perf/payment-apps-map' of github.com:calcom/cal.com into perf/payment-apps-map

keithwillcode committed:
Removed build commands that were for testing

keithwillcode committed:
Fix filename issue

joeauyeung committed:
Type fixes

keithwillcode committed:
fixed type issue

keithwillcode committed:
Merge branch 'perf/payment-apps-map' of github.com:calcom/cal.com into perf/payment-apps-map

keithwillcode committed:
Committed the wrong file

zomars commented on packages/app-store/_utils/getCalendar.ts:
```diff
@@ -36,20 +25,16 @@ export const getCalendar = async (
+  ];
 
-  if (!isCalendarService(calendarApp)) {
+  if (!calendarService.default) {
```
Seems like we have a type error in here

hbjORbj committed:
Merge branch 'main' into perf/payment-apps-map

joeauyeung committed:
Fix CalendarService import

joeauyeung committed:
Upgrade vitest to 3.0.8

joeauyeung committed:
Merge branch 'main' into perf/payment-apps-map

joeauyeung committed:
Remove app store e2e and add routing form e2e

joeauyeung committed:
Fix routing form yaml import

joeauyeung committed:
Replace any instance of e2e-app-store with e2e-routing-forms

joeauyeung committed:
Fix typo

joeauyeung committed:
Revert playwright version bump

joeauyeung committed:
Fix mock imports

hbjORbj committed:
Merge remote-tracking branch 'origin/main' into perf/payment-apps-map

hbjORbj committed:
remove campsite

hbjORbj committed:
Merge remote-tracking branch 'origin/main' into perf/payment-apps-map

keithwillcode committed:
Remove unwanted file

joeauyeung committed:
At runtime mock calendar integration - not working

keithwillcode committed:
Merge branch 'main' into perf/payment-apps-map

sean-brydon committed:
correct app-store/mock

keithwillcode committed:
Fixed video adapters mock

keithwillcode committed:
Removed dynamic imports to test

keithwillcode committed:
Merge remote-tracking branch 'origin/main' into perf/payment-apps-map

keithwillcode committed:
Merge remote-tracking branch 'origin/main' into perf/payment-apps-map

keithwillcode committed:
Merge remote-tracking branch 'origin/main' into perf/payment-apps-map

keithwillcode committed:
Removed dyanmic app store imports

keithwillcode committed:
Fixed imports

keithwillcode committed:
Merge branch 'main' into perf/payment-apps-map

keithwillcode committed:
fix issues from merge

devin-ai-integration[bot] committed:
fix: update app-store-cli to generate static imports

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update mock implementations in bookingScenario.ts

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update mock implementations in bookingScenario.ts to work with map-based imports

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: finalize mock implementations for map-based imports

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update app-store-cli to properly generate static imports

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update app-store-cli to generate static imports instead of dynamic imports

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: add static imports for EventTypeAppSettingsInterface components

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: remove dynamic imports from browser component files

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: add static imports to server.generated.ts and ensure all browser files use static imports

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update app-store-cli to generate proper static imports and fix import references

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update appStoreMetadata import path in mockCalendar.ts

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update VideoApiAdapter instantiation in videoClient.ts

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update logger import in mockCalendar.ts

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update import paths in bookingScenario.ts

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update VideoApiAdapter instantiation to use constructor

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update VideoApiAdapter instantiation in videoClient.ts

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update CalendarService and VideoApiAdapter instantiation

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update CalendarService instantiation in getCalendar.ts

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update CalendarService instantiation to use constructor

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update CalendarService instantiation with proper class reference

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: simplify CalendarService instantiation in getCalendar.ts

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update app-store-cli build.ts to use static imports

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update CalendarService and VideoApiAdapter instantiation

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update VideoApiAdapter instantiation to use function call instead of constructor

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: remove dynamic import from apps.browser-eventtypesettings.generated.tsx

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update CalendarService instantiation to use function call instead of constructor

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update import paths in build.ts to use lib directory

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: handle hyphenated property names in generated files

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update import syntax in generated files to use default imports

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update PaymentService and VideoApiAdapter instantiation

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update import syntax in CRM and payment services

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update import syntax in calendar and video adapter services

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update calendar service instantiation and build.ts import patterns

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update CalendarService instantiation to use constructor instead of function call

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update import patterns in build.ts to use named exports

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update import paths in generated files to use specific file paths

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update import statements in build.ts to use default imports

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update build.ts to use named exports for service imports

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update build.ts to use default imports for service files

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update import patterns in generated files to use correct export patterns

Co-Authored-By: keith@cal.com <keith@cal.com>

keithwillcode committed:
Fixed types

devin-ai-integration[bot] committed:
fix: update MockCalendarService to use calEvent parameter instead of event

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: remove require statements in bookingScenario.ts

Co-Authored-By: keith@cal.com <keith@cal.com>

keithwillcode committed:
Merge branch 'main' into perf/payment-apps-map

devin-ai-integration[bot] committed:
fix: update mock functions to properly track method calls

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update mock implementations to track method calls and fix type issues

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: update interface definitions to make externalCalendarId optional

Co-Authored-By: keith@cal.com <keith@cal.com>

devin-ai-integration[bot] committed:
fix: resolve TypeScript type errors in booking fields transformer and location handling

- Fix LocationDefaultFieldInput_2024_06_14 type compatibility issues with defensive property checks
- Add missing createdAt/updatedAt fields to destinationCalendar objects in buildCalEventFromBooking
- Update editLocation handler to properly handle user destinationCalendar timestamps
- Ensure type-safe transformations for booking field inputs with optional properties

Co-Authored-By: keith@cal.com <keithwillcode@gmail.com>

keithwillcode commented:
Closing for now since we couldn't get the mocking sorted. We know this implementation tremendously helps so will continue exploring with #22630 and #22450
